modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -179,16 +186,25 @@ public AsyncIndexBySearchAction(BulkByScrollTask task, ESLogger logger, ParentTa
         @Override
         protected ScrollableHitSource buildScrollableResultSource(BackoffPolicy backoffPolicy) {
             if (mainRequest.getRemoteInfo() != null) {
-                // NORELEASE track 500-level retries that are builtin to the client
                 RemoteInfo remoteInfo = mainRequest.getRemoteInfo();
+                HttpClientBuilder httpClient = HttpClients.custom();
                 if (remoteInfo.getUsername() != null) {
-                    // NORELEASE support auth
-                    throw new UnsupportedOperationException("Auth is unsupported");
+                    UsernamePasswordCredentials creds = new UsernamePasswordCredentials(remoteInfo.getUsername(), remoteInfo.getPassword());
+                    CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                    credentialsProvider.setCredentials(AuthScope.ANY, creds);
+                    httpClient.setDefaultCredentialsProvider(credentialsProvider);
                 }
-                RestClient restClient = RestClient.builder(new HttpHost(remoteInfo.getHost(), remoteInfo.getPort(), remoteInfo.getScheme()))
-                        .build();
+                Header[] clientHeaders = new Header[remoteInfo.getHeaders().size()];
```

Comment:
Headers are nice for working with proxies and required for the testing solution I've set up.